### DISCUSSION TITLE
Support for aleatoric and other non-static compositional structures?

### DISCUSSION THREAD

**Discussion by ahankinson** [music-encoding/music-encoding#889] [Ideas]
_From [a...@kb.dk](https://code.google.com/u/101822190470135414035/) on July 30, 2013 04:48:40_

Some ossia- or choice-like element would be useful to encode non-fixed or optional contents like aleatoric pieces ("musikalische Würfelspiele" etc.).
The options in this case are given by the composer, but the choice is to be made by the performer during or prior to the performance (thus it is not an editorial decision, so editorial elements like <choice> and <rdg> will not do). 
Something like ossia at a higher level than 1 measure/1 staff?

_Original issue: http://code.google.com/p/music-encoding/issues/detail?id=170_

**Comment by ahankinson** [comment]
_From [pd...@virginia.edu](https://code.google.com/u/103686026181985548448/) on June 11, 2014 13:30:27_

**Labels:** -Priority-Medium Priority-Low

**Comment by craigsapp** [comment]
For musikalische Würfelspiele, each measure could be encoded in a separate `<section>`, and then expansion labels can be generated to render a particular performance sequence:

<img width="590" alt="Screen Shot 2022-01-24 at 6 37 58 AM" src="https://user-images.githubusercontent.com/3487289/150803368-e35aca2d-f4ca-45b5-99b1-7a8980f05c63.png">

Below is an encoding of the above score, using

```xml
      <expansion plist="#label-B #label-C #label-A" xml:id="expansion-L3F1" />
```

to indicate the performance sequence of measure 2, 3, then 1.

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="https://music-encoding.org/schema/dev/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="https://music-encoding.org/schema/dev/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="5.0.0-dev">
 <meiHead>
  <fileDesc>
   <titleStmt>
    <title />
   </titleStmt>
   <pubStmt />
  </fileDesc>
  <encodingDesc>
   <appInfo>
    <application isodate="2022-01-24T06:36:55" version="3.9.0-dev-a25f001">
     <name>Verovio</name>
     <p>Transcoded from Humdrum</p>
    </application>
   </appInfo>
  </encodingDesc>
  <workList>
   <work>
    <title />
   </work>
  </workList>
 </meiHead>
 <music>
  <body>
   <mdiv xml:id="mjnwqm8">
    <score xml:id="senobtk">
     <scoreDef xml:id="sd2x0x3" midi.bpm="400.000000">
      <staffGrp xml:id="s5avsu5">
       <staffDef xml:id="s8ehlqh" n="1" lines="5">
        <clef xml:id="c1ukmwj" shape="G" line="2" />
        <meterSig xml:id="metersig-L2F1" count="4" unit="4" />
       </staffDef>
      </staffGrp>
     </scoreDef>
     <section xml:id="section-L1F1">
      <expansion plist="#label-B #label-C #label-A" xml:id="expansion-L3F1" />
      <section xml:id="label-A">
       <measure xml:id="measure-L1" n="1">
        <staff xml:id="sqm4ccs" n="1">
         <layer xml:id="layer-L1F1N1" n="1">
          <note xml:id="note-L6F1" dur="1" oct="4" pname="c" accid.ges="n" />
         </layer>
        </staff>
       </measure>
      </section>
      <section xml:id="label-B">
       <measure xml:id="measure-L8" n="2">
        <staff xml:id="staff-L8F1N1" n="1">
         <layer xml:id="layer-L8F1N1" n="1">
          <note xml:id="note-L9F1" dur="1" oct="4" pname="d" accid.ges="n" />
         </layer>
        </staff>
       </measure>
      </section>
      <section xml:id="label-C">
       <measure xml:id="measure-L10" right="end" n="3">
        <staff xml:id="staff-L10F1N1" n="1">
         <layer xml:id="layer-L10F1N1" n="1">
          <note xml:id="note-L12F1" dur="1" oct="4" pname="e" accid.ges="n" />
         </layer>
        </staff>
       </measure>
      </section>
     </section>
    </score>
   </mdiv>
  </body>
 </music>
</mei>
```

But for this sort of algorithmic score, it would be better to use XSLT to render the performance sequence score by rearranging/selecting measures in the original score.

See related:   https://dice.humdrum.org and https://verovio.humdrum.org/?file=dice/mozart/k516f.krn

Here is a particular realization:

https://verovio.humdrum.org/?file=dice/mozart/k516f.krn&k=ey&filter=myank%20-m%20%2745%2C154%2C86%2C62%2C33%7C%7C16%2C88%2C19%2C175%2C73%2C136%2C23%2C172%3D%3D%27

Where the sequence of measure numbers is given in the URL:

```
45,154,86,62,33,16,88,19,175,73,136,23,172
```
